kymatio/scattering2d/frontend/base_frontend.py

```diff
@@ -17,24 +17,34 @@ def __init__(self, J, shape, L=8, max_order=2, pre_pad=False,
         self.out_type = out_type
 
     def build(self):
-        self.M, self.N = self.shape
+        M, N = self.shape
 
-        if 2 ** self.J > self.M or 2 ** self.J > self.N:
+        if 2 ** self.J > M or 2 ** self.J > N:
             raise RuntimeError('The smallest dimension should be larger than 2^J.')
-        self.M_padded, self.N_padded = compute_padding(self.M, self.N, self.J)
+        self._M_padded, self._N_padded = compute_padding(M, N, self.J)
```

Comment:
It might be good for this to give us a tuple containing the padded shape rather than having two seperate variable for ot.